programs/manifest/src/quantities.rs

```diff
@@ -325,23 +325,25 @@ impl QuoteAtomsPerBaseAtom {
         }
     }
 
-    pub fn multiply_spread(self, spread_e_5: u32) -> Self {
+    pub fn multiply_spread(self, spread_e_decimals: u32, decimals: u32) -> Self {
         // Stored as u128 * 10^-26
         let inner: u128 = u64_slice_to_u128(self.inner);
-        let inner_e_minus_5: u128 = inner.wrapping_mul(spread_e_5 as u128);
-        let new_inner: u128 = inner_e_minus_5.div_ceil(100_000);
+        // multiply then divide
+        // p * (spread / 10^decimals)
+        let inner_e_decimals: u128 = inner.wrapping_mul(spread_e_decimals as u128);
+        let new_inner: u128 = inner_e_decimals.div_ceil(10_u128.pow(decimals) as u128);
         QuoteAtomsPerBaseAtom {
             inner: u128_to_u64_slice(new_inner),
         }
     }
 
-    pub fn divide_spread(self, spread_e_5: u32) -> Self {
+    pub fn divide_spread(self, spread_e_decimals: u32, decimals: u32) -> Self {
         // multiply then divide
         QuoteAtomsPerBaseAtom {
             inner: u128_to_u64_slice(
                 u64_slice_to_u128(self.inner)
-                    .wrapping_mul(100_000)
-                    .div_ceil(spread_e_5 as u128),
+                    .wrapping_mul(10_u128.pow(decimals))
```

Comment:
use lookup table in quantities

Reply:
wrapping_mul with div behind it is not a good idea here, it can silently truncate the price on the MSB, in that case better to truncate on the LSB